common/check_expired_locked_rules

```diff
@@ -6,51 +6,96 @@
-        print 'Locked expired rules'
-        for row in session.execute(query):
+        print('Locked expired rules')
+        query = (session.query(models.ReplicationRule.id, models.ReplicationRule.scope, 
+        models.ReplicationRule.name, models.ReplicationRule.rse_expression))
+        .filter(models.ReplicationRule.locked == 1)
+        .filter(models.ReplicationRule.expires_at != None)
+        .filter(models.ReplicationRule.expires_at < datetime.utcnow())
+        results = query.all()
+        for row in results:
             status = CRITICAL
-            print row[0], row[1], row[2]
+            id = unicode(row[0])
+            print(id, row[1], row[2])
+        print('Locked expired rule counts by RSE expression')
+        query = (session.query(models.ReplicationRule.scope, models.ReplicationRule.rse_expression,
+                               func.count(models.ReplicationRule.scope))
+                 .filter(models.ReplicationRule.locked == 1, models.ReplicationRule.expires_at <= datetime.utcnow())
+                 .group_by(models.ReplicationRule.scope, models.ReplicationRule.rse_expression))
+        for row in query.all():
+            print('judge.locked_expired_rules.%s.%s    %s' % (row[0], row[1], row[2]))
+            monitor.record_gauge(stat='judge.locked_expired_rules.%s.%s' % (row[0], row[1]), value=row[2])
     except Exception as error:
-        print error
+        print(error)
         status = UNKNOWN
         sys.exit(status)
+
     try:
-        query = """select rawtohex(c.id), c.scope, c.name, c.rse_expression from atlas_rucio.rules c,
-                   (select a.scope, a.name from atlas_rucio.dids a
-                   where a.expired_at is not null and a.expired_at < sys_extract_utc(localtimestamp)
-                   and exists (select 1 from atlas_rucio.rules b where a.scope=b.scope and a.name=b.name and locked=1)) d
-                   where c.scope=d.scope and c.name=d.name and locked=1"""
-        print 'Datasets expired with locked rules'
-        for row in session.execute(query):
+        print('Datasets expired with locked rules')
+        query = (session.query(models.ReplicationRule.id,models.ReplicationRule.scope,
+        models.ReplicationRule.name,models.ReplicationRule.rse_expression)
+        .join(models.DataIdentifier, (models.ReplicationRule.scope == models.DataIdentifier.scope)
+                           & (models.ReplicationRule.name == models.DataIdentifier.name))
+                     .filter(models.ReplicationRule.locked == True)
+                     .filter(models.DataIdentifier.expired_at != None)
+                     .filter(models.DataIdentifier.expired_at < datetime.utcnow()))
+        for row in query.all():
             status = CRITICAL
-            print row[0], row[1], row[2], row[3]
+            id = unicode(row[0])
+            print(id, row[1], row[2], row[3])
```

Comment:
@dchristidis do we gain anything by printing out the results of these queries? CMS is only using this to feed to monitoring, so we don't care about it.

Reply:
Printing is on the old probes, so I just copied that. Should I delete those? @dchristidis